server/handler/base.go

```diff
@@ -193,5 +206,58 @@ func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, cl
 		return errors.Errorf("evaluation resulted in unexpected state: %s", result.Status)
 	}
 
+	if statusState == "pending" && prctx.IsDraft() != nil && !*prctx.IsDraft() {
```

Comment:
This probably needs to be moved out into a better place

Reply:
In particular, we may want to only run this on `pull_request` events to avoid duplicate/conflicting evaluations. It would also be nice if this logic was located somewhere it could be tested in isolation (like the approval rules); maybe we could split the part that finds and selects reviewers into a different package and have it return a list of reviewers that are assigned here?